ros/behaviors/chat/scripts/chat_node.py

```diff
@@ -0,0 +1,596 @@
+
+        # WARNING : This is a workaround to wait for the service to be done
+        # It is not possible to use rclpy.spin_until_future_complete because
+        # it will block the executor and the node will not be able to process
+        # other callbacks even when using a ReentrantCallbackGroup and a MultiThreadedExecutor.
+        # For this to work, the service must be called in a separate thread.
+        event = Event()
+
+        def service_done_cb(future):
+            event.set()
+
+        future.add_done_callback(service_done_cb)
+        event.wait(timeout=5.0)
+
+        if future.done() and future.result() is not None:
+            self.get_logger().info(f"Service call result: {future.result()}")
+            return future.result()
+
+        # Something went wrong
+        self.get_logger().error(f"Service call failed: {future.exception()}")
+        return None
+
+    def add_pending_message(self, message: str):
+        self._pending_messages.append(message)
+        self._process_pending_messages()
+
+    def _on_transcript_received_cb(self, msg: Transcript):
+        self.get_logger().info(f"Transcript received: {msg.text}")
+
+        self._talking = False
+        self._processing = False
+
+        if len(msg.text) > 0:
+            # Add the transcript to the context history
+            self._chat_api.add_to_history(
+                message=msg.text, role="user", timestamp=datetime.now()
+            )
+
+            # Process the request
+            self._processing = True
+            self.get_logger().info("Processing...")
+            self._chat_api.send_request_and_process_response()
+            self._processing = False
+            self.get_logger().info("Processing done!")
+
+        else:
+            self.get_logger().error("Empty transcript")
+
+        # Safety always call _process_pending_messages
+        self._process_pending_messages()
+
+    def _on_talk_done_cb(self, msg: Done):
+        self.get_logger().info(f"Talk done : {msg.ok}")
+        self._talking = False
+        self._process_pending_messages()
+
+    def _process_pending_messages(self):
+        if not self._talking:
+            partial_message: str = ""
+            for message in self._pending_messages:
+                partial_message += message
+            self._pending_messages.clear()
+
+            # To be done we need to have no pending messages
+            if len(partial_message) == 0:
+                # We are done
+                chat_msg = Done()
+                chat_msg.ok = True
+                self.get_logger().info("Chat done")
+                self._chat_done_pub.publish(chat_msg)
+                return
+
+            talk_msg = Text()
+
+            # Remove all the text between the <think> </think> tags
+            # This is present in thinking models
+            partial_message = re.sub(
+                r"<think>.*?</think>", "", partial_message, flags=re.DOTALL
```

Comment:
I think this is redundant: `*` already means "0 or more".
```suggestion
                r"<think>.*</think>", "", partial_message, flags=re.DOTALL
```